src/test/java/org/wso2/lsp4intellij/utils/FileUtilsTest.java

```diff
@@ -0,0 +1,200 @@
+
+    Assert.assertEquals(editor, FileUtils.editorFromVirtualFile(file, project));
+
+    Assert.assertNull(FileUtils.editorFromVirtualFile(file, project));
+  }
+
+  @PrepareForTest(LocalFileSystem.class)
+  @Test
+  public void testVirtualFileFromURI() {
+    LocalFileSystem localFileSystem = PowerMockito.mock(LocalFileSystem.class);
+    PowerMockito.mockStatic(LocalFileSystem.class);
+    PowerMockito.when(LocalFileSystem.getInstance()).thenReturn(localFileSystem);
+    PowerMockito.when(LocalFileSystem.getInstance().findFileByIoFile(Mockito.any()))
+        .thenReturn(new BinaryLightVirtualFile());
+
+    Assert.assertEquals(new BinaryLightVirtualFile().toString(),
+        FileUtils.virtualFileFromURI("file://foobar").toString());
+  }
+
+  @PrepareForTest(LocalFileSystem.class)
+  @Test
+  public void testVirtualFileFromURINull() {
+    PowerMockito.mockStatic(LocalFileSystem.class);
+    PowerMockito.when(LocalFileSystem.getInstance()).thenThrow(URISyntaxException.class);
+
+    Assert.assertNull(FileUtils.virtualFileFromURI("foobar"));
+  }
+
+  @Test
+  public void testVFSToURI() {
+    VirtualFile virtualFile = PowerMockito.mock(VirtualFile.class);
+    PowerMockito.when(virtualFile.getUrl()).thenReturn("file://fooBar");
+
+    Assert.assertEquals("file:///fooBar", FileUtils.VFSToURI((virtualFile)));
+  }
+
+  @Test
+  public void testVFSToURINull() {
+    Assert.assertNull(FileUtils.VFSToURI((new LightVirtualFile())));
+  }
+
+  @Test
+  public void testSanitizeURIUnix() {
+    PowerMockito.mockStatic(System.class);
+    PowerMockito.when(System.getProperty(Mockito.anyString())).thenReturn("Linux");
+
+    Assert.assertNull(FileUtils.sanitizeURI(null));
+
+    Assert.assertEquals("fooBar", FileUtils.sanitizeURI("fooBar"));
+    Assert.assertEquals("file:///fooBar", FileUtils.sanitizeURI("file://fooBar"));
+  }
+
+  @PrepareForTest(FileUtils.class)
+  @Test
+  public void testSanitizeURIWindows() {
+    PowerMockito.mockStatic(System.class);
+    PowerMockito.when(System.getProperty(Mockito.anyString())).thenReturn("Windows");
+
+    Assert.assertEquals("file:///Foo:/Bar", FileUtils.sanitizeURI("file:foo%3A/Bar"));
+  }
+
+  @PrepareForTest(LocalFileSystem.class)
+  @Test
+  public void testURIToVFS() {
+    LocalFileSystem localFileSystem = PowerMockito.mock(LocalFileSystem.class);
+    PowerMockito.mockStatic(LocalFileSystem.class);
+    PowerMockito.when(LocalFileSystem.getInstance()).thenReturn(localFileSystem);
+    PowerMockito.when(LocalFileSystem.getInstance().findFileByIoFile(Mockito.any()))
+        .thenReturn(new BinaryLightVirtualFile());
+
+    Assert.assertEquals(new BinaryLightVirtualFile().toString(),
+        FileUtils.URIToVFS("file://foobar").toString());
+  }
+
+  @PrepareForTest(LocalFileSystem.class)
+  @Test
+  public void testURIToVFSNull() {
+    PowerMockito.mockStatic(LocalFileSystem.class);
+    PowerMockito.when(LocalFileSystem.getInstance()).thenThrow(URISyntaxException.class);
```

Comment:
and this one?

Reply:
same as above :
this is done to reach a line in a try catch statement which only catches this specific exception.